<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SetupWizard</class>
 <widget class="QWizard" name="SetupWizard">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>480</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>640</width>
    <height>480</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Setup</string>
  </property>
  <property name="wizardStyle">
   <enum>QWizard::ModernStyle</enum>
  </property>
  <widget class="QWizardPage" name="gosumemory">
   <property name="title">
    <string>Gosumemory setup</string>
   </property>
   <property name="subTitle">
    <string>Welcome to miraya setup wizard</string>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <property name="topMargin">
     <number>9</number>
    </property>
    <property name="verticalSpacing">
     <number>20</number>
    </property>
    <item row="1" column="0">
     <layout class="QFormLayout" name="gosumemoryFormLayout">
      <property name="horizontalSpacing">
       <number>20</number>
      </property>
      <property name="verticalSpacing">
       <number>20</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="gosumemoryIpLabel">
        <property name="text">
         <string>IP Address</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="gosumemoryPortLabel">
        <property name="text">
         <string>Port</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="gosumemoryIpLineEdit">
        <property name="inputMethodHints">
         <set>Qt::ImhUrlCharactersOnly</set>
        </property>
        <property name="text">
         <string>127.0.0.1</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="gosumemoryPortLineEdit">
        <property name="inputMethodHints">
         <set>Qt::ImhDigitsOnly</set>
        </property>
        <property name="text">
         <string>24050</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="0" colspan="2">
     <widget class="QLabel" name="gosumemoryExplanationLabel">
      <property name="contextMenuPolicy">
       <enum>Qt::NoContextMenu</enum>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;First off, let's set up the connection between osu and miraya. &lt;/p&gt;&lt;p&gt;To do that, download &lt;a href=&quot;https://github.com/l3lackShark/gosumemory/releases/latest&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#007af4;&quot;&gt;gosumemory&lt;/span&gt;&lt;/a&gt; and start it. &lt;/p&gt;&lt;p&gt;If you are planning to use miraya on the pc that is running gosumemory and osu, press next to continue the setup.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
      <property name="openExternalLinks">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="osuapi">
   <property name="title">
    <string>osu! API</string>
   </property>
   <property name="subTitle">
    <string>This will help miraya to grab data from beatmaps when they are requested.</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="2" column="0" rowspan="2">
     <layout class="QFormLayout" name="osuapiFormLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="osuapiClientIdLabel">
        <property name="text">
         <string>osu! Client ID</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="osuapiClientIdLineEdit">
        <property name="inputMethodHints">
         <set>Qt::ImhDigitsOnly</set>
        </property>
        <property name="maxLength">
         <number>6</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="osuapiClientSecretLabel">
        <property name="text">
         <string>osu! Client Secret</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="osuapiClientSecretLineEdit">
        <property name="inputMethodHints">
         <set>Qt::ImhHiddenText|Qt::ImhNoAutoUppercase|Qt::ImhNoPredictiveText|Qt::ImhSensitiveData</set>
        </property>
        <property name="echoMode">
         <enum>QLineEdit::Password</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="0" colspan="2">
     <widget class="QLabel" name="osuapiExplanationLabel">
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Go to your &lt;a href=&quot;https://osu.ppy.sh/home/account/edit&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#4285f4;&quot;&gt;osu! account settings&lt;/span&gt;&lt;/a&gt; and create a new OAuth application (at the bottom of the page). &lt;/p&gt;&lt;p&gt;You can chose whatever name you want, and there is no need to put any application callback URL.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="twitch">
   <property name="title">
    <string>Twitch Setup</string>
   </property>
   <property name="subTitle">
    <string>This is used to read the messages from your chat and, eventually, to respond to them</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_3">
    <property name="verticalSpacing">
     <number>20</number>
    </property>
    <item row="1" column="0">
     <layout class="QFormLayout" name="twitchFormLayout">
      <property name="horizontalSpacing">
       <number>20</number>
      </property>
      <property name="verticalSpacing">
       <number>20</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="twitchBotNickLabel">
        <property name="text">
         <string>Bot Nickname</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="twitchBotNickLineEdit"/>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="twitchOauthLineEdit">
        <property name="echoMode">
         <enum>QLineEdit::Password</enum>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="twitchChannelLineEdit"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="twitchOauthLabel">
        <property name="text">
         <string>OAuth</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="twitchChannelLabel">
        <property name="text">
         <string>Channel Name</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="twitchExplanationLabel">
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Usually, a Twitch chatbot uses a different account from the streamer.&lt;/p&gt;&lt;p&gt;1. Create a twitch account for your bot.&lt;/p&gt;&lt;p&gt;2. Log into it and go to &lt;a href=&quot;https://twitchapps.com/tmi/&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#007af4;&quot;&gt;https://twitchapps.com/tmi/&lt;/span&gt;&lt;/a&gt; to request a oauth code. This will be needed to connect Miraya to Twitch.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
      <property name="openExternalLinks">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="osuirc">
   <property name="title">
    <string>osu IRC setup</string>
   </property>
   <property name="subTitle">
    <string>Osu IRC is the in-game osu chat. This will be used to send you the maps in game directly from twitch chat.</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_5">
    <property name="verticalSpacing">
     <number>20</number>
    </property>
    <item row="1" column="0">
     <layout class="QFormLayout" name="osuircFormLayout">
      <property name="horizontalSpacing">
       <number>20</number>
      </property>
      <property name="verticalSpacing">
       <number>20</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="osuircNickLabel">
        <property name="text">
         <string>Nickname</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="osuircPasswordLabel">
        <property name="text">
         <string>Password</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="osuircServerLabel">
        <property name="text">
         <string>Server</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="osuircPortLabel">
        <property name="text">
         <string>Port</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="osuircNickLineEdit"/>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="osuircPasswordLineEdit">
        <property name="echoMode">
         <enum>QLineEdit::Password</enum>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="osuircServerLineEdit">
        <property name="inputMethodHints">
         <set>Qt::ImhUrlCharactersOnly</set>
        </property>
        <property name="text">
         <string>irc.ppy.sh</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="osuircPortLineEdit">
        <property name="inputMethodHints">
         <set>Qt::ImhDigitsOnly</set>
        </property>
        <property name="text">
         <string>6667</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="osuircExplanationLabel">
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Osu! uses a communication system called IRC to supply the in-game chat. &lt;/p&gt;&lt;p&gt;Miraya can connect to it and send map requests from you Twitch chat effortlessly!&lt;/p&gt;&lt;p&gt;To do that, go to &lt;a href=&quot;osu.ppy.sh/p/irc&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#007af4;&quot;&gt;osu.ppy.sh/p/irc&lt;/span&gt;&lt;/a&gt;, login and copy the nickname and password.&lt;/p&gt;&lt;p&gt;If you'd like to use Miraya in custom servers, you can change the IRC server address.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="openExternalLinks">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="filesharing">
   <property name="title">
    <string>Filesharing setup</string>
   </property>
   <property name="subTitle">
    <string>This is used to automatically upload your skins and beatmaps</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="0" column="0">
     <widget class="QLabel" name="filesharingExplanationLabel">
      <property name="text">
       <string>TBD</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
